packages/wrangler/src/deployment-bundle/source-maps.ts

```diff
@@ -0,0 +1,136 @@
+import type { CfModule, CfWorkerSourceMap } from "./worker";
+
+/**
+ * Loads source maps that appear in the given build output.
+ */
+export function loadSourceMaps(
+	main: CfModule,
+	modules: CfModule[],
+	bundle: Partial<BundleResult>
+): CfWorkerSourceMap[] {
+	const { sourceMapPath, sourceMapMetadata } = bundle;
+	if (sourceMapPath && sourceMapMetadata) {
+		// This worker was bundled by Wrangler, so we already know where
+		// the source map is located.
+		return loadSourceMap(main, sourceMapPath, sourceMapMetadata);
+	} else {
+		// Don't know where source maps are located, so need to find
+		// them by scanning the contents of the user-specified modules.
+		return scanSourceMaps([main, ...modules]);
+	}
+}
+
+interface SourceMapV3 {
+	version: 3;
+	file: string;
+	sourceRoot: string;
+	sources: string[];
+	sourcesContent: string[];
+	mappings: string;
+	names: string[];
+}
+
+/**
+ * Load and normalize a source map emitted by Wrangler using the given path and
+ * directory metadata.
+ */
+function loadSourceMap(
+	{ name, filePath }: CfModule,
+	sourceMapPath: string,
+	{ entryDirectory }: SourceMapMetadata
+): CfWorkerSourceMap[] {
+	if (filePath === undefined) {
+		return [];
+	}
+	const map = JSON.parse(
+		fs.readFileSync(sourceMapPath).toString()
+	) as SourceMapV3;
+	// Overwrite the file property of the source map to match the name of the
+	// main module in the multipart upload.
+	map.file = name;
+	// Remove the temporary directory prefix generated by Wrangler that appears
+	// in the source root path.
+	const sourceRootPath = path.dirname(path.join(entryDirectory, sourceMapPath));
+	map.sourceRoot = stripPrefix(sourceRootPath, map.sourceRoot);
+	map.sources = map.sources.map((source) => {
+		const originalPath = path.join(path.dirname(filePath), source);
+		// Remove the temporary build directory prefix generated by Wrangler
+		// that appears in the source path.
+		return stripPrefix(withTrailingSep(entryDirectory), originalPath);
+	});
+	return [
+		{
+			name: name + ".map",
+			content: JSON.stringify(map),
+		},
+	];
+}
+
+/**
+ * Find source maps by scanning module contents for special `//#
+ * sourceMappingURL=` comments pointing to source map files.
+ */
+function scanSourceMaps(modules: CfModule[]): CfWorkerSourceMap[] {
+	const maps: CfWorkerSourceMap[] = [];
+	for (const module of modules) {
+		const commentPrefix = "//# sourceMappingURL=";
+		const lastLine = module.content.toString().split("\n").pop();
+		if (
+			lastLine === undefined ||
+			!lastLine.startsWith(commentPrefix) ||
+			module.filePath === undefined
+		) {
+			continue;
+		}
+		// Assume the source map path in the comment is relative to the
+		// generated file it appears in.
+		const commentPath = stripPrefix(commentPrefix, lastLine).trim();
+		if (commentPath.startsWith("data:")) {
+			throw new Error(
+				`Invalid source map path in ${module.filePath}: expected file path but found data URL.`
```

Comment:
Strictly this is not invalid, it is just unsupported by this code.

Reply:
Updated the error message to say "unsupported" instead of "invalid"